llvm/lib/Target/RISCV/RISCVMCInstLower.cpp

```diff
@@ -31,7 +31,7 @@ static MCOperand lowerSymbolOperand(const MachineOperand &MO, MCSymbol *Sym,
   MCContext &Ctx = AP.OutContext;
   RISCVMCExpr::VariantKind Kind;
 
-  switch (MO.getTargetFlags()) {
+  switch (MO.getTargetFlags() & ~RISCVII::MO_JUMP_TABLE_BASE) {
```

Comment:
Shouldn't this be the direct flag mask, otherwise it will break once we add another bitflag?
```suggestion
  switch (MO.getTargetFlags() & MO_DIRECT_FLAG_MASK) {
```

Reply:
I view it the other way, like this we catch flags here that we're not (yet) expecting. Checking only `& MO_DIRECT_FLAG_MASK` would weaken the error check.

Reply:
Sounds good to me.